src/api/app/models/relationship/add_role.rb

```diff
@@ -0,0 +1,48 @@
+class Relationship::AddRole
+  class SaveError < APIError; end
+
+  def initialize(package_or_project, role, opts)
+    self.package_or_project = package_or_project
+    self.role = role
+    self.check = opts[:check]
+
+    package_or_project.check_write_access!(opts[:ignore_lock])
+    check_role!
+    check_add_role_arguments!(opts)
+  end
+
+  def add_role
+    return if duplicate?
+    relationship = package_or_project.relationships.build(user: user, group: group, role: role)
+    relationship.delete if relationship.invalid?
+  end
+
+  private
+
+  attr_accessor :package_or_project, :role, :user, :group, :check
+
+  def duplicate?
+    return unless package_or_project.relationships.where(user: user, group: group, role: role).exists?
+    raise SaveError, 'Relationship already exists' if check
+    debug_str = user ? "user #{user.login}" : "group #{group.title}"
```

Comment:
can you remove this debug code, please? I know it was already there, but if you could remove it would be great!

Reply:
why?

Reply:
because to have a debug here is arbitrary. Why this method and not the others? We should have a way to introspect but to have arbitrary debugging is a smell.

Reply:
hmm, that's true for the other 140 debug statements as well. So I consider this a nitpick - and if would have been easy to mention it in your first review. So I don't want to reopen this

Reply:
fair enough.